ls1intum/Artemis
# Exam-mode/client - remaining time in start view

### Checklist

### Motivation and Context


The students should be able to download the exam earlier than at exam start in order to minimize the workload of the server. They also should be told that this is the right place to be and how long they need to wait.
### Description

This feature adds an overlay so the students know when the exam is going to start. 
Now, the "start Exam" button depends on the visibiltyDate, not the startDate anymore.


### Steps for Testing

- Locally, create a new exam
- Make sure the visibilityDate ist before the current date and the startDate is after the current date
- If you go to the exam and check the checkmark with the confirmation text, the startExam button should be clickable
- click the start Exam button, an overlay should be displayed
- Make sure the time until exam start is correct and the startDate of the exam afterwards is displayed correctly
- If the remaining time is below 3 minutes and 30 seconds, those seconds should also be displayed


### Screenshots

![Quiz](https://user-images.githubusercontent.com/44401560/85198259-85c9b600-b2e7-11ea-9c13-05b21daad4e7.JPG)

![Capture2](https://user-images.githubusercontent.com/44401560/85198143-81e96400-b2e6-11ea-8067-578b1fbb6ff2.JPG)
This is the current endscreen. Next step would be, when the actually exam starts, to change "waitingForExamStart" to false again so the overlay vanishes and the exam begins
